src/components/CodeInput/index.js

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import HelpText from '../Input/styled/helpText';
+import InputItems from './inputItems';
+import RenderIf from '../RenderIf';
+import RequiredAsterisk from '../RequiredAsterisk';
+import StyledTextError from '../Input/styled/errorText';
+import { useReduxForm, useUniqueIdentifier } from '../../libs/hooks';
+import { useFocusedIndexState, useValueState, usePreviousIndex } from './hooks';
+import { getNormalizedValue, getNumbersFromClipboard, setFocus } from './helpers';
+import { StyledFieldset, StyledLabel } from './styled';
+
+/**
+ * The CodeInput is an element that allows to fill a list of numbers, suitable for code validations.
+ * @category Form
+ */
+const CodeInput = React.forwardRef((props, ref) => {
+    const {
+        id,
+        name,
+        value: valueProp,
+        label,
+        bottomHelpText,
+        length,
+        disabled,
+        required,
+        readOnly,
+        error,
+        tabIndex,
+        onClick,
+        onChange,
+        onFocus,
+        onBlur,
+        onKeyDown,
+        className,
+        style,
+    } = useReduxForm(props);
+
+    const value = useValueState(valueProp, length);
+    const focusedIndex = useFocusedIndexState(value, length);
+    const previousFocusedIndex = usePreviousIndex(focusedIndex);
+    const inputId = useUniqueIdentifier('code-input');
+
+    useEffect(() => {
+        if (previousFocusedIndex !== undefined) {
+            setFocus(ref);
+        }
+    }, [ref, focusedIndex, previousFocusedIndex]);
+
+    const handleOnChange = (inputValue, inputIndex) => {
+        onChange(getNormalizedValue(inputValue, inputIndex, value));
+    };
+
+    const handleOnFocus = (e, index) => {
+        if (focusedIndex !== index) {
+            setFocus(ref);
+        }
+        onFocus(e);
+    };
+
+    const handleOnPaste = e => {
+        onChange(getNumbersFromClipboard(e.clipboardData.getData('Text')));
+    };
+
+    return (
+        <StyledFieldset className={className} name={name} style={style} id={id}>
+            <RenderIf isTrue={!!label}>
+                <StyledLabel htmlFor={inputId}>
```

Comment:
In this case, the input id does not get associated with the label because it is a legend element and not a label.